README.md

```diff
@@ -87,6 +88,87 @@ To skip or include multiple step names, separate them with space, like in this e
 dabs.sh -c examples/apps/hello-world-app --skip-steps test_unit test_performance
 ```
 
+## Detailed execution steps
+
+`abs` is composed of two main pipelines: *build* and *test*. Each of them is composed of steps.
+When `abs` runs, it executes all the steps from the *build* pipeline and then from the *test* pipeline.
+Config options can be used to disable/enable any specific build steps.
+
+Please check below for available build pipelines and steps. Each step offers some config options,
+you can check them by running `dabs.sh -h`.
+
+### Build pipelines
+
+Currently, only one build pipeline is supported. It is based on `helm 3`.
+
+#### Helm 3 build engine steps
+
+Helm 3 build pipeline executes in sequence the following set of steps:
+
+1. HelmBuilderValidator: a simple step that checks if the build folder contains a Helm chart
+1. HelmGitVersionSetter: when enabled, this step will set `version` and/or `appVersion` in the `Chart.yaml`
+   of your helm chart to a version value based of your last commit hash and tag in a git repo. For this
+   step to work, the chart or chart's parent directory must contain valid git repo (`.git/`).
+1. HelmChartToolLinter: this step runs the [`ct`](https://github.com/helm/chart-testing) (aka. `chart-testing`)
+   This tool runs validation and linting of YAML files included in your chart.
+1. HelmChartMetadataPreparer: this step is required to gather some data required for chart metadata
+   generation.
+1. HelmChartBuilder: this step does the actual chart build using Helm.
+1. HelmChartMetadataFinalizer: completes and writes the data gather partially by HelmChartMetadataPreparer.
+1. HelmChartYAMLRestorer: restores chart files, which were changed as part of the build process (ie. by
+   HelmGitVersionSetter).
+
+### Test pipelines
+
+After your app artifact is built (your chart when using Helm build engine pipeline), `abs` can run
+tests for it for you. There are a few assumptions related to how testing invoked by `abs` works.
+
+First, we assume that each test framework that you can use for developing tests for your app can
+label the tests and run only the set of tests labelled. `abs` expects all tests to have at least one
+of the following labels: `smoke`, `functional`.
```

Comment:
Needs an explanation (or a link to an explanation or example) of `labels`.

Reply:
Is it clearer now?